src/worksheet.rs

```diff
@@ -7697,14 +7698,10 @@ impl Worksheet {
     }
 
     // Cached/faster version of utility.col_to_name() to use in the inner loop.
-    fn col_to_name(&mut self, col_num: ColNum) -> String {
-        if let Some(col_name) = self.col_names.get(&col_num) {
-            col_name.clone()
-        } else {
-            let col_name = utility::col_to_name(col_num);
-            self.col_names.insert(col_num, col_name.clone());
-            col_name
-        }
+    fn col_to_name(col_names: &mut HashMap<u16, String>, col_num: ColNum) -> &str {
```

Comment:
Is there a performance benefit to converting this from a method to an associated function? If not, and this is a semantic change, then I'd prefer it to stay as a method. If it improves performance then the change is okay.

Reply:
`col_to_name` is a hot function. Without this change, we would need to clone the `&str` in order to use `&mut self.writer.xmlfile` afterwards. Those clones (allocations) had a measurable difference and showed up in flamegraph.

Reply:
Ok. +1 to this change in that case.

Reply:
One other possible optimisation is that for columns <= 25 we could have a shortcut conversion from the index to the ASCII char like this code from utility.rs:

```rust
let col_letter = char::from_u32(64u32 + remainder as u32).unwrap();
```

Or simpler still a indexed char slice of a str like `"ABCDEF...XYZ"`.

Reply:
Good idea for a future PR